File: python3/reverse_string.py




def reverse_string(inp):
    inp = inp.split(' ')
    for word in inp:
        wlen = len(word) - 1
        for i in range(wlen):
            word[i], word[wlen - i] = word[wlen - i], word[i]
    return " ".join(inp)

# Testing:
inp = "Let's take LeetCode contest"
out = "s'teL ekat edoCteeL tsetnoc"

reverse_string(inp)
print(out == inp)

File: c/reverse_string.c
/*
 * Input: s = "Let's take LeetCode contest"
 * Output: "s'teL ekat edoCteeL tsetnoc"
 * */

#include <assert.h>


char* reverse_string(char*);

int main() {
	char* in = "Let's take LeetCode contest";
	char* out = "s'teL ekat edoCteeL tsetnoc";
	reverse_string(&in);
	assert(in == out);
}

char* reverse_string(char*) {
	
}
